packages/framework-core/src/booster-token-verifier.ts

```diff
@@ -54,12 +55,20 @@ class TokenVerifierClient {
         if (err) {
           return reject(err)
         }
-        return resolve(this.tokenToUserEnvelope(decoded))
+        const jwtToken = decoded as any
+        if (this.tokenVerifierConfig?.extraValidation) {
+          try {
+            this.tokenVerifierConfig?.extraValidation(jwtToken?.header, jwtToken?.payload)
+          } catch (err) {
+            reject(err)
+          }
+        }
+        return resolve(this.tokenPayloadToUserEnvelope(jwtToken?.payload))
```

Comment:
Why not add the header to the user envelope too? I don't know the reason at the moment, but it could be useful for security logging and it's cheap to add it.

```suggestion
        return resolve(this.tokenToUserEnvelope(jwtToken))
```

Reply:
Fixed in d43ea9cb9848e447a39abaa483e34388613d0524